NUS-Fintech-Society/Atlas_2
# ATL-6 Create Event

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Part 3 (Latest: 16/2/23):
- Create hamburger navbar at the top of create events page with additional sign out functionality

Part 2:
- Integrate backend with new db schema. For the creation of `event`, connects existing `Departments` and `User` data upon creation of an `event`.

Part 2 Notes:
- Created some data in the db for `Departments` so that I can reference upon creating an event.
- Used uuid for `eventId` 

Part 2 Future Improvements:
- In the schema e.g the`Departments` model has a `relation` with `Event` but it is currently 1-to-1. So in the event that we create a new `event` with departments that are already tied to an event, existing event tied to the department will be overriden and only the latest event is shown. Could improve to a 1-to-many in the future. This is also the same case for `User` as it is currently a 1-to-1 relationship with `Event`
- How do we create columns in railway that are of our custom type we have defined e.g Departments[] property for event

Part 1
- Create event frontend based off Evan's work, added form logic and validation
- Add create event backend that creates an event and saves to db, based off ziyuan's work

Part 1 Notes:
- Form validation for department checkboxes for scenario when submitting no selected departments still displays default error message by zod not very sure how to change it, must be due to it submitting an empty array: `[]` which is identified as `boolean` thus displaying message "Expected array, received boolean"
- Form logic and validation for `attendees` is abit hacky since I had to manually create it instead of integrating with `react-hook-forms`, as the `IndeterminateCheckbox` component for the `DataTable` requires an `onChange`  for its toggle functionality. This clashes with react-hook-forms' `register` function that implements its own `onChange`, such that I cannot register `attendees` into `react-hook-forms` 
- Typing is not very robust at the moment

Fixes # (issue)
ATL-6

## Type of change

Please delete options that are not relevant.


# Checklist:
